### PR TITLE
Unityswitchon

### PR DESCRIPTION
**Pull request by davidoae**
environments/production/hiera/common.json_unityswitch contains example config change, I did think about removing everything else that doesn't change or even not including it but it's nice to have an example to look at.

### CODE REVIEW

**Comment by davidoae** [comment]
Forgot to say, there is an associated doc here;
https://github.com/oaeproject/Hilary/wiki/Domain-Switch-and-Redirects

File: modules/nginx/manifests/redirect.pp [hunk 1]
```diff
@@ -0,0 +1,44 @@
+define nginx::redirect (
+    $web_domain		= hiera("web_domain"),
+    $web_domain_redirect_from = hiera("web_domain_redirect_from"),
+    $ssl_crt_source     = "puppet:///modules/localconfig/ssl/*.${web_domain_redirect_from}/server.crt",
+    $ssl_key_source     = "puppet:///modules/localconfig/ssl/*.${web_domain_redirect_from}/server.key",
+    $server_name        = $name,
```
Comment (mrvisser) [inline modules/nginx/manifests/redirect.pp]: `server_name` sounds more something like `pp0`, `app1`, but I think this is more intended to the "tenant alias" in OAE terminology. Can you rename it to something like `tenant_alias`?

Comment (davidoae) [inline modules/nginx/manifests/redirect.pp]: Hmm, server_name is the same terminology used elsewhere, eg in server.pp
which sets up web_domains_external, which is why I used the same here.

Comment (mrvisser) [inline modules/nginx/manifests/redirect.pp]: That's fair enough. Seems I've been out of the game for too long :)

File: modules/nginx/manifests/redirect.pp [hunk 2]
```diff
@@ -0,0 +1,44 @@
+define nginx::redirect (
+    $web_domain		= hiera("web_domain"),
```
Comment (simong) [inline modules/nginx/manifests/redirect.pp]: We don't usually let non-oaeservice modules pull from hiera. I'd prefer it if the oaeservice/manifests/nginx.pp module would pass this information in.

Comment (mrvisser) [inline modules/nginx/manifests/redirect.pp]: :+1: Good catch

File: environments/production/hiera/common.json [hunk 3]
```diff
@@ -45,8 +45,10 @@
     },
     "deploy_user_passwd": "$6$KIxK1V/a$HpQSDSjkeh9BKsPWnFJ41pojj3ndVUJlkal8o.dpPxWHLxZYFo44OmwKVj7cm8.enpaiBodlNZ8ix/l2Wm0Ge.",
 
-    "web_domain": "oaeproject.org",
-    "web_domains_external": ["collab.lib.cam.ac.uk", "oae.gatech.edu", "oae.its.txstate.edu", "shib-sp.unity.ac", "open.vu.nl"],
+    "web_domain": "unity.ac",
+    "web_domains_external": ["*.oaeproject.org", "collab.lib.cam.ac.uk", "oae.gatech.edu", "oae.its.txstate.edu", "shib-sp.unity.ac", "open.vu.nl"],
+    "web_domains_redirect": ["testtenancy"],
```
Comment (simong) [inline environments/production/hiera/common.json]: I assume this needs to be `research` ?

File: modules/nginx/manifests/redirect.pp [hunk 4]
```diff
@@ -0,0 +1,42 @@
+define nginx::redirect (
+    $ssl_crt_source     = "puppet:///modules/localconfig/ssl/*.${web_domain_redirect_from}/server.crt",
```
Comment (simong) [inline modules/nginx/manifests/redirect.pp]: I think you may have removed the `$web_domain` and `$web_domain_redirect_from` here by accident

**Comment by simong** [comment]
I've fixed those issues on staging and it deploys correctly. I'm happy with rolling this out on production .

**Pull request merged**